foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java

```diff
@@ -197,11 +197,23 @@ private void getProperties(ConfigurableEnvironment environment, PropertySource<?
     if (propertySource instanceof EnumerablePropertySource) {
       EnumerablePropertySource<?> enumerablePropertySource = (EnumerablePropertySource<?>) propertySource;
       for (String propertyName : enumerablePropertySource.getPropertyNames()) {
-        configFromSpringBoot.put(propertyName, environment.getProperty(propertyName, Object.class));
+        try {
+          configFromSpringBoot.put(propertyName, environment.getProperty(propertyName, Object.class));
+        } catch (Exception e) {
+          if (!getIfIgnoreEnvironment()) {
+            throw new RuntimeException("set up spring property source failed.", e);
+          } else {
+            LOGGER.warn("set up spring property source failed.", e);
+          }
+        }
       }
       return;
     }
 
     LOGGER.debug("a none EnumerablePropertySource is ignored, propertySourceName = [{}]", propertySource.getName());
   }
+
+  private boolean getIfIgnoreEnvironment() {
+    return (Boolean) ConfigUtil.createLocalConfig().getProperty("servicecomb.config.ignoreResolveFailure");
```

Comment:
NPE would be arised if you haven't set any value for  servicecomb.config.ignoreResolveFailure